src/app/fdctl/run/tiles/fd_dedup.c

```diff
@@ -212,32 +211,8 @@ unprivileged_init( fd_topo_t *      topo,
                    fd_topo_tile_t * tile ) {
   void * scratch = fd_topo_obj_laddr( topo, tile->tile_obj_id );
 
-  /* Frankendancer has gossip_dedup, verify_dedup+
-     Firedancer has gossip_dedup, voter_dedup, verify_dedup+ */
-  ulong unparsed_in_cnt = 1;
-  if( FD_UNLIKELY( tile->in_cnt<2UL ) ) {
-    FD_LOG_ERR(( "dedup tile needs at least two input links, got %lu", tile->in_cnt ));
-  } else if( FD_UNLIKELY( strcmp( topo->links[ tile->in_link_id[ GOSSIP_IN_IDX ] ].name, "gossip_dedup" ) ) ) {
-    /* We have one link for gossip messages... */
-    FD_LOG_ERR(( "dedup tile has unexpected input links %lu %lu %s",
-                 tile->in_cnt, GOSSIP_IN_IDX, topo->links[ tile->in_link_id[ GOSSIP_IN_IDX ] ].name ));
-  } else {
-    /* ...followed by a voter_dedup link if it were the Firedancer topology */
-    ulong voter_dedup_idx = fd_topo_find_tile_in_link( topo, tile, "voter_dedup", 0 );
-    if( voter_dedup_idx!=ULONG_MAX ) {
-      FD_TEST( voter_dedup_idx == VOTER_IN_IDX );
-      unparsed_in_cnt = 2;
-    } else {
-      unparsed_in_cnt = 1;
-    }
-
-    /* ...followed by a sequence of verify_dedup links */
-    for( ulong i=unparsed_in_cnt; i<tile->in_cnt; i++ ) {
-      if( FD_UNLIKELY( strcmp( topo->links[ tile->in_link_id[ i ] ].name, "verify_dedup" ) ) ) {
-        FD_LOG_ERR(( "dedup tile has unexpected input links %lu %lu %s",
-                     tile->in_cnt, i, topo->links[ tile->in_link_id[ i ] ].name ));
-      }
-    }
+  if( FD_UNLIKELY( tile->in_cnt == 0UL ) ) {
```

Comment:
I wouldn't even bother to check this, topology constructor can do what they like.

Reply:
sure